packages/ra-core/src/controller/field/useReferenceArrayFieldController.spec.tsx

```diff
@@ -166,4 +166,379 @@ describe('<useReferenceArrayFieldController />', () => {
             })
         );
     });
+
+    describe('displaySelectAllButton', () => {
+        it('should return true if no items are selected', async () => {
+            const children = jest.fn().mockReturnValue('child');
+            render(
+                <CoreAdminContext dataProvider={dataProvider}>
+                    <ReferenceArrayFieldController
+                        resource="foo"
+                        reference="bar"
+                        record={{ id: 1, barIds: [1, 2] }}
+                        source="barIds"
+                    >
+                        {children}
+                    </ReferenceArrayFieldController>
+                </CoreAdminContext>
+            );
+            await waitFor(() => {
+                expect(children).toHaveBeenNthCalledWith(
+                    1,
+                    expect.objectContaining({
+                        displaySelectAllButton: true,
+                        data: undefined,
+                        total: undefined,
+                    })
+                );
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenNthCalledWith(
+                    2,
+                    expect.objectContaining({
+                        displaySelectAllButton: true,
+                        data: undefined,
+                        total: undefined,
+                    })
+                );
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenNthCalledWith(
+                    3,
+                    expect.objectContaining({
+                        displaySelectAllButton: true,
+                        data: [
+                            { id: 1, title: 'bar1' },
+                            { id: 2, title: 'bar2' },
+                        ],
+                        total: 2,
+                    })
+                );
+            });
```

Comment:
Aren't we just interested by this? Same for many other checks
```suggestion
            await waitFor(() => {
                expect(children).toHaveBeenCalledWith(
                    expect.objectContaining({
                        displaySelectAllButton: true,
                        data: [
                            { id: 1, title: 'bar1' },
                            { id: 2, title: 'bar2' },
                        ],
                        total: 2,
                    })
                );
            });
```